.github/workflows/cnb.yml

```diff
@@ -0,0 +1,99 @@
+  actions: write
+  contents: write
+  pull-requests: write
+
+jobs:
+  update-cnb:
+    name: Update .NET CNB version
+    runs-on: pub-hk-ubuntu-24.04-ip
+
+    steps:
+      - uses: actions/create-github-app-token@v1
+        id: generate-token
+        with:
+          app-id: ${{ vars.LINGUIST_GH_APP_ID }}
+          private-key: ${{ secrets.LINGUIST_GH_PRIVATE_KEY }}
+
+      - name: Checkout Repo
+        uses: actions/checkout@v4
+        with:
+          token: ${{ steps.generate-token.outputs.token }}
+
+      - name: Get next release and update files
+        id: get_next_release
+        env:
+          GH_TOKEN: ${{ steps.generate-token.outputs.token }}
+        run: |
+          # Get the currently pinned version
+          previous_version=$(awk -F'"' '/DOTNET_CNB_VERSION=/ {print $2}' bin/compile)
+
+          # Get all releases sorted by creation date
+          releases=$(gh release list --repo heroku/buildpacks-dotnet --limit 100 --json tagName --jq '.[].tagName' | tac)
+          next_release=""
+
+          # Find the next release after the pinned version
+          found_previous=false
+          for release in $releases; do
+            if [ "$found_previous" = "true" ]; then
+              next_release=${release#v}
+              break
+            fi
+            if [ "$release" = "v$previous_version" ]; then
+              found_previous=true
+            fi
+          done
+
+          # Handle cases where no next release is found
+          if [ -z "$next_release" ]; then
+            echo "No newer release found. Exiting."
+            exit 0
+          fi
+
+          echo "next_release=$next_release" >> $GITHUB_ENV
+
+          # Fetch release notes
+          RELEASE_NOTES=$(gh release view "v$next_release" --repo heroku/buildpacks-dotnet --json body --jq .body)
+          SAFE_NOTES=$(echo "$RELEASE_NOTES" | awk '/^### / {p=1} p {print}' | sed '/^## /,$d')
+
+          # Update bin/compile
+          sed -i "s/DOTNET_CNB_VERSION=\".*\"/DOTNET_CNB_VERSION=\"$next_release\"/" bin/compile
+
+          # Update CHANGELOG.md
+          if ! grep -q "## \[Unreleased\]" CHANGELOG.md; then
+            echo -e "## [Unreleased]\n\n$(cat CHANGELOG.md)" > CHANGELOG.md
+          fi
+          awk -v notes="$SAFE_NOTES" '/^## \[Unreleased\]/ {print; print ""; print notes; next} 1' CHANGELOG.md > temp && mv temp CHANGELOG.md
+
+      - name: Create Pull Request
+        id: create_pr
+        uses: peter-evans/create-pull-request@v7.0.5
+        with:
+          token: ${{ steps.generate-token.outputs.token }}
+          title: "Update heroku/buildpacks-dotnet to v${{ env.next_release }}"
+          commit-message: "Update DOTNET_CNB_VERSION version to ${{ env.next_release }}"
+          committer: ${{ vars.LINGUIST_GH_APP_USERNAME }} <${{ vars.LINGUIST_GH_APP_EMAIL }}>
+          author: ${{ vars.LINGUIST_GH_APP_USERNAME }} <${{ vars.LINGUIST_GH_APP_EMAIL }}>
+          branch: update-dotnet-cnb-${{ env.next_release }}
```

Comment:
(optional)

Using the version in the branch name means separate PRs will be opened for different versions. This may be intentional given the "version+1 update strategy here" (in which case no problem), but for other repos we've intentionally used a fixed branch name, since the `create-pull-request` action is smart enough to update any existing PR should a new version become available over time etc.